### PR TITLE
HTS-1097: Implemented get account API

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosaveapi/connectors/HelpToSaveConnector.scala [hunk 1]
```diff
@@ -51,11 +53,19 @@ class HelpToSaveConnectorImpl @Inject() (config: Configuration,
...
+  val acceptHeaderName: String = config.underlying.getString("microservice.acceptHeaderName")
+
   override def createAccount(body: CreateAccountBody, correlationId: UUID)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] =
     http.post(createAccountUrl, body, Map(correlationIdHeaderName -> correlationId.toString))
 
   override def checkEligibility(nino: String, correlationId: UUID)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] =
     http.get(eligibilityCheckUrl(nino), Map(correlationIdHeaderName -> correlationId.toString))
+
+  override def getAccount(nino: String, correlationId: UUID)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] =
+    http.get(getAccountUrl, Map(correlationIdHeaderName -> correlationId.toString, acceptHeaderName → "application/vnd.hmrc.2.0+json"))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosaveapi/connectors/HelpToSaveConnector.scala]: I don't think we need the accept header value here in the request to the BE

File: app/uk/gov/hmrc/helptosaveapi/controllers/HelpToSaveAuth.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import uk.gov.hmrc.auth.core.AuthProvider.GovernmentGateway
+import uk.gov.hmrc.auth.core._
+import uk.gov.hmrc.auth.core.authorise.Predicate
+import uk.gov.hmrc.auth.core.retrieve.Retrievals
+import uk.gov.hmrc.helptosaveapi.util.{Logging, NINO, WithMdcExecutionContext, toFuture}
+import uk.gov.hmrc.play.bootstrap.controller.BaseController
+
+import scala.concurrent.Future
+
+object HelpToSaveAuth {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosaveapi/controllers/HelpToSaveAuth.scala]: There's already stuff in `https://github.com/hmrc/help-to-save-api/blob/master/app/uk/gov/hmrc/helptosaveapi/auth/Auth.scala` which you could re-use

File: app/uk/gov/hmrc/helptosaveapi/services/HelpToSaveApiService.scala [hunk 3]
```diff
@@ -172,21 +197,33 @@ class HelpToSaveApiServiceImpl @Inject() (helpToSaveConnector: HelpToSaveConnect
...
       )
   }
 
+  private def validateGetAccountRequest(f: () ⇒ GetAccountResponseType)(implicit request: Request[_]): GetAccountResponseType = {
+    httpHeaderValidator.validateHttpHeaders(false).toEither
+      .fold[GetAccountResponseType](
+        e ⇒ {
+          val errorString = s"[${e.toList.mkString("; ")}]"
+          logger.warn(s"Error when validating get account request, errors: $errorString")
+          Left(ApiError("SERVER_ERROR", "There was a server error"))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosaveapi/services/HelpToSaveApiService.scala]: when validation fails we should be returning a 400 (bad request)

File: app/uk/gov/hmrc/helptosaveapi/models/EligibilityCheckError.scala [hunk 4]
```diff
@@ -20,7 +20,7 @@ import play.api.libs.json.{Format, JsValue, Json}
 
 sealed trait EligibilityCheckError
 
-case class EligibilityCheckValidationError(code: String, message: String) extends EligibilityCheckError
+case class EligibilityCheckValidationError(apiError: ApiError) extends EligibilityCheckError
 
 object EligibilityCheckValidationError {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosaveapi/models/EligibilityCheckError.scala]: we don't need the JSON formats for `EligibilityCheckValidationError` and `EligibilityCheckBackendError` - we shouldn't be serialising those case classes to JSON. We should only be serialising the underlying `ApiError`

File: app/uk/gov/hmrc/helptosaveapi/connectors/HelpToSaveConnector.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/connectors/HelpToSaveConnector.scala]: Ok I have removed it.

File: app/uk/gov/hmrc/helptosaveapi/controllers/HelpToSaveAuth.scala (hunk 2 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/controllers/HelpToSaveAuth.scala]: I have removed HelpToSaveAuth as I was already using Auth anyway.

File: app/uk/gov/hmrc/helptosaveapi/services/HelpToSaveApiService.scala (hunk 3 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosaveapi/services/HelpToSaveApiService.scala]: ok within the ApiError?

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**